### PR TITLE
Add typing information for operation decorator

### PR DESCRIPTION
**Pull request by iwoloschin**
This is a quick PR to simply add typing information for the `@operation` decorator. With this typing information added it allows pylance/pyright (and presumably mypy & pyre, though I haven't tested those) to properly check decorated operations. It does *not* check for global arguments (such as `name`), those require additional typing definitions to be added, that would be handled in a subsequent PR.

I have not done any testing beyond on my own system (python3.8 on macOS) so consider this a bit of an exploratory PR. Happy to add/change things as needed.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -119,6 +119,7 @@
             'ansible': ANSIBLE_REQUIRES,
         },
         include_package_data=True,
+        package_data={'pyinfra': ['py.typed', 'api/*.pyi']},
```
Comment (Fizzadar) [inline setup.py:122]: I believe this should be in `MANIFEST.in` rather than here, they'll override the manifest file currently!

**Review by Fizzadar — COMMENTED** [review]
@iwoloschin thanks for the PR! This looks great, awesome to see type annotations coming into the project! One change w/packaging stuff.

I'm going to create a branch specifically to build out type annotations & setup mypy requirement/checking as part of CI/etc, and will change the PR base branch to that.

File: setup.py (hunk 1 above)
Comment (iwoloschin) [reply setup.py:122]: Ah, I've been using Poetry myself lately, so I wasn't really sure how to handle that. Easy enough to change.

**Comment by blarghmatey** [comment]
One thing that might be helpful in retroactively adding type information is https://google.github.io/pytype/. It will scan the existing code and infer types based on static analysis which might speed up some of the initial work.

**Review by Fizzadar — APPROVED** [review]
Finally got the base branch + testing setup here! Thank  you for the PR @iwoloschin :)

**Comment by iwoloschin** [comment]
Awesome! I'll see if I can start adding some more typing things in. I'll probably focus on things I'm using myself first, but if there are any specific things that you think would be helpful to focus on let me know.

**Pull request merged**